core/trino-main/src/main/java/io/trino/execution/resourcegroups/InternalResourceGroupManager.java

```diff
@@ -248,6 +249,17 @@ private synchronized void createGroupIfNecessary(SelectionContext<C> context, Ex
             configurationManager.get().configure(group, context);
             checkState(groups.put(id, group) == null, "Unexpected existing resource group");
         }
+        else if (currentGroup.isDisabled()) {
```

Comment:
nit: I would move this to the previous commit `Add flag to mark resource groups as disabled` as this looks like a tightly coupled change and it seems adding a flag without this logic is a no-op